### ISSUE TITLE
Uncaught ReferenceError: getNetworkIPAddress is not defined

### ISSUE THREAD

**Issue by jackejiang** [willemmulder/Diont-for-Cordova#2]
I ran in android, and get following message:

I/chromium(10190): [INFO:CONSOLE(100)] "Uncaught ReferenceError: getNetworkIPAddress is not defined", source: file:///android_asset/www/plugins/net.willemmulder.diont.cordova/www/diont.js (101)

getNetworkIPAddress is a method of nodejs os module. Anythong wrong?

**Comment by jackejiang** [comment]
getNetworkIPAddress is a method which depends on the nodejs os module in the non-cordova version diont.js

**Comment by willemmulder** [comment]
Right, you're correct. I guess I never published services on mobile, but only consumed them ;-)

What should happen is that a call is made to the Java part and a function there returns the proper IP address. 

Would anybody be able to take a stab at this?

**Comment by Samardon** [comment]
I decided to try using WebRTC to get the local network address and everything seems to be working (emulator).

I submitted a pull request with the changes I made, but would greatly appreciate it if someone would be kind enough to double-check that this holds water, as I'm not an expert.

**Comment by willemmulder** [comment]
Hi Samardon,

WebRTC should work. It will hold its external IP address somewhere, because it will need to expose that to its other connections. Did it work in your app?

Regardless, WebRTC support is only supported on android browser 37 and Chrome. So a Java solution that works on older phones as well, might be worth a shot as well.

**Comment by Samardon** [comment]
It worked as expected, but not as intended.

I also couldn't agree more about the need to be able to support older phones.

I submitted a much more sensible pull request where I've added MobileChromeApp's chrome.system.network Plugin as a dependency and simply leveraged their API to grab the host address.

I've tested it on a real device and thus far it's gone off without a hitch.

**Comment by willemmulder** [comment]
Hi Samardon, excellent!
I've merged your pull request, many thanks.

@jackejiang can you confirm this solves your problem as well? Then I'll close the issue.

**Comment by Samardon** [comment]
Glad to help!
